addons/mail/controllers/webclient.py

```diff
@@ -8,9 +8,9 @@
 
 class WebclientController(http.Controller):
     @http.route("/mail/init_messaging", methods=["POST"], type="json", auth="public")
-    def mail_init_messaging(self):
+    def mail_init_messaging(self, **kwargs):
         if not request.env.user.sudo()._is_public():
-            return request.env.user.sudo(request.env.user.has_group("base.group_portal"))._init_messaging()
```

Comment:
Can't you immediately parse `allowed_company_ids` from here and set it in the real context rather than forwarding the param everywhere?

This would ease the diff and the risk of the PR by not changing API of methods too much.